### PR TITLE
use main lodash instead of separate packages per function

### PR DESCRIPTION
**Pull request by gazs**
One of our low-hanging fruits is the duplication of lodash imports. Our codebase currently has 4 ways and I'm trying to get it down to 2 (see https://github.com/conde-nast-international/voguede-autopilot/pull/823 ). This is an attempt to fix the upstream instead of trying to rewrite everything on our end.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,8 +22,7 @@
   "license": "MIT",
   "dependencies": {
     "jsonml.js": "^0.1.0",
-    "lodash.camelcase": "^4.1.1",
-    "lodash.isfunction": "^3.0.8"
+    "lodash": "^4.17.4"
```
Comment (pgoldrbx) [inline package.json]: probably fine, though we could just specify `^4` to allow maximum deduping downstream?

Comment (pgoldrbx) [reply package.json]: Ok, i've done a search, and the lowest common version is `4.14.1`

https://github.com/search?l=JSON&p=2&q=org%3ACondeNast+jsonmltoreact&type=Code

- [autopilot-ad](https://github.com/CondeNast/autopilot-ad/blob/a2434fdb29f5e72356214eb511f2adc3b0a038d2/package.json)
- [autopilot-teen-vogue](https://github.com/CondeNast/autopilot-teen-vogue/blob/527ccb72c7ace773c897bd0961ec5af1281cbc89/package.json)
- [autopilot-self](https://github.com/CondeNast/autopilot-self/blob/86894083ba950c995979b11e7755b6833f55343a/package.json)

cc @dcolucci

**Comment by giles-v** [comment]
So one of the nice things about this lib is the small size. You're probably juicing it by 15kB or something with this change. Is the assumption that brand apps will _always_ be using the full lodash?

**Comment by pgoldrbx** [comment]
@giles-v right - the trouble is we either need to _always_ use only the exact lodash packages needed, or we need to always use the full, everywhere, so that we can dedupe in the bundle. If we have a combination, the result will always be larger.

I think there's another thread somewhere about this

**Comment by brokentone** [comment]
This is excellent, thank you @gazs -- thought I can't see the referenced repo.

@giles-v -- yes I think we have to make this assumption at this point -- and theoretically, it still could be treeshaken downstream. With different packages, it cannot be.

**Comment by gazs** [comment]
@giles-v this would only increase the build size for brand apps that didn't use lodash at all or were only using the `lodash.foo` named modules. If they were using `lodash/foo`, this should be a size reduction for them, if they are importing the monolithical `lodash`, this should be a size-neutral change (getting rid of `lodash.foo` and the other lodashmodules it brings in )

**Comment by rufus2021** [comment]
could this could result in two versions of the same method for some apps?

**Comment by rharriso** [comment]
@rufus2021 it could, but even if it isn't resolved by npm, you can force a single version of lodash with webpack

**Comment by mize** [comment]
Aside from James' timing question, would `peerDependencies` be useful here? I know we've used the trick in other libs of adding a package as both a `peerDependency` (when used by a dependent lib/app) and a `devDependency` (when used for this lib's local dev).

**Comment by pgoldrbx** [comment]
Yeah, I've considered peerDependencies as a solution as well

**Comment by pgoldrbx** [comment]
Still blocked on the bundle size change?

**Comment by pgoldrbx** [comment]
This got lost a long time ago, but really it should have landed. Let's make it land!

**Comment by rharriso** [comment]
:+1: 

Looks like I'm still part of some team. Which is good, I'm planning on using this in a project coming up here.

**Comment by diffcunha** [comment]
Good to know this is still being used! 👍

**Comment by ajkim** [comment]
💪

**Comment by mf2492** [comment]
💯

**Comment by JulieBusch** [comment]
I see I'm not the only one who was mildly surprised to see a Conde email
from Github. I applaud your efforts, Phil!

**Comment by bencmilton** [comment]
giphy

**Comment by gkilian** [comment]
Nothing like a Conde reunion inside a PR!!!

**Comment by bmullan91** [comment]
Good lad Phil 💪

**Comment by brokentone** [comment]
I thought more lodash was always better. E.g. at Condé I made a brute force object property extractor leveraging lodash! https://gist.github.com/brokentone/9639f8e0a073aec66ae3b3c67cdf5d9a

**Comment by taveras** [comment]
👏 👏

**Comment by Juan-M** [comment]
Wait for me!!!

**Comment by poscar** [comment]
👋

**Comment by Georgette** [comment]
GeGe is still here.

**Comment by Georgette** [comment]
ngmz0

**Comment by fesebuv** [comment]
Well done sir! @pgoldrbx

**Comment by whyisjake** [comment]
Well done, everyone.

**Pull request merged**